framework/src/outputs/formatters/JSONFormatter.C

```diff
@@ -0,0 +1,168 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "JSONFormatter.h"
+#include "Parser.h"
+#include "MooseEnum.h"
+
+#include <sstream>
+#include <vector>
+
+JSONFormatter::JSONFormatter(bool dump_mode) :
+    SyntaxTree(true),
+    _dump_mode(dump_mode)
+{
+}
+
+std::string
+JSONFormatter::postscript() const
+{
+  std::stringstream s;
+  s << "**START JSON DATA**\n" << _json << "\n**END JSON DATA**\n";
+  return s.str();
+}
+
+std::vector<std::string> JSONFormatter::splitPath(const std::string& long_name) const
```

Comment:
Technically these ampersands are binary operators and should have spacing on both side. There are a ton of them that are not formatted correctly in MOOSE today but new code should follow the standard.